mapper/tedge_mapper/src/component.rs

```diff
@@ -0,0 +1,7 @@
+use async_trait::async_trait;
+use tedge_config::TEdgeConfig;
+
+#[async_trait]
+pub trait TEdgeComponent {
+    async fn start(&self, tedge_config: TEdgeConfig) -> Result<(), anyhow::Error>;
```

Comment:
@ChrisGreenaway is the plan to add "stop", signal handling etc. to that later on?

Reply:
I did think about the stop case and - because it's a Future, you can just stop polling it.

I hadn't thought too much about the signal side of things. That's not in the current code and I'm not adding new functionality, just refactoring. But I could imagine adding a `reload()` method, or similar. But equally, I had thought that dropping the existing Future and starting another would restart this component.